src/components/UserList.js

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react';
+import Search from './Search';
+import User from './User';
+
+
+class UserList extends Component {
+   
+    state = {
+        search: '',
+        loading: true,
+        users: []
+    };
+    componentWillReceiveProps(props, state) {
```

Comment:
It's fine for your practice and remembrance of the order of lifecycle hooks, but in real life cases, we don't declare them unless we need any of those.